dnsapi/dns_ispconfig.sh

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env sh
+
+#ISPConfig 3.1 API - Add remote user and give him access to at least the "DNS txt functions"
+
+ISPC_User=""
+ISPC_Password=""
+# Provide proper URL and port for your ISPC Installation
+ISPC_Api="https://ispc.domain.tld:8080/remote/json.php"
+
+########  Public functions #####################
+
+#Usage: dns_myapi_add   _acme-challenge.www.domain.com   "XKrxpRBosdIKFzxW_CT3KLZNf6q0HG9i01zxXp5CPBs"
+dns_ispconfig_add() {
+  fulldomain="${1}"
+  txtvalue="${2}"
+  _ISPC_login
+  if [ $? -eq 0 ]; then
+    _ISPC_getZoneInfo
+  fi
+  if [ $? -eq 0 ]; then
+    _ISPC_addTxt
+  fi
+  if [ $? -ne 0 ]; then
+    return 1
+  fi
+}
+
+#Usage: dns_myapi_rm   _acme-challenge.www.domain.com
+dns_ispconfig_rm() {
+  fulldomain="${1}"
+  _ISPC_login
+  if [ $? -eq 0 ]; then
+    _ISPC_rmTxt
+  fi
+  if [ $? -ne 0 ]; then
+    return 1
+  fi
+}
+
+####################  Private functions bellow ##################################
+
+_ISPC_login() {
+  _info "Getting Session ID"
+  curData="{\"username\":\"${ISPC_User}\",\"password\":\"${ISPC_Password}\",\"client_login\":false}"
+  curResult=$(curl -k --data "${curData}" "${ISPC_Api}?login")
+  if _contains "${curResult}" '"code":"ok"'; then
+    sessionID=$(echo "${curResult}" | _egrep_o "response.*" | cut -d ':' -f 2)
+    sessionID=${sessionID:1:-2}
+    _info "Successfully retrieved Session ID."
+  else
+    _err "Couldn't retrieve the Session ID."
+  fi
+}
+
+_ISPC_getZoneInfo() {
+  _info "Getting Zoneinfo"
+  zoneEnd=false
+  curZone="${fulldomain}"
+  while [ ${zoneEnd} = false ]; do
+    # we can strip the first part of the fulldomain, since it's just the _acme-challenge string
+    curZone="${curZone#*.}"
+    # suffix . needed for zone -> domain.tld.
+    curData="{\"session_id\":\"${sessionID}\",\"primary_id\":[{\"origin\":\"${curZone}.\"}]}"
+    curResult=$(curl -k --data "${curData}" "${ISPC_Api}?dns_zone_get")
```

Comment:
Problem is, that no -k option is being sent. ISPConfig a selfhosted control panel and by default it will create its own certs which are not trusted by browsers or curl by default.